pyomo/common/tests/test_dependencies.py

```diff
@@ -233,6 +234,20 @@ def test_import_exceptions(self):
         self.assertFalse(avail)
         self.assertFalse(avail)
 
+        mod, avail = attempt_import('pyomo.common.tests.dep_mod_except',
+                                    defer_check=True,
+                                    catch_exceptions=(ImportError, ValueError))
+        self.assertFalse(avail)
+        self.assertFalse(avail)
```

Comment:
Repeated assert?

Reply:
Yes.  One of the magic things that attempt_import does is that the returned objects actually replace themselves with the resolved values when the deferred resolution actually happens.  This implicitly checks both the original return value, and the repeated return value.